File: 백준/Bronze/25304. 영수증/영수증.js
const fs = require('fs');

const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const total = parseInt(input[0]);
const num = input[1];
let testTotal = 0;

for (let i = 0; i < num; i++) {
  const test = input[i + 2].split(' ').map(Number);
  testTotal += test[0] * test[1];
}

total === testTotal ? console.log('Yes') : console.log('No');

File: 백준/Bronze/25304. 영수증/영수증.py
import sys

total_price = int(sys.stdin.readline().strip())
total_num = int(sys.stdin.readline().strip())
res = 0

for i in range(1, total_num + 1):
  price, num = list(map(int, sys.stdin.readline().strip().split()))
  res += price * num
  
print('Yes') if res == total_price else print('No')